packages/keyboard-shortcuts/src/hooks/use-shortcut.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * External dependencies
+ */
+import Mousetrap from 'mousetrap';
+import 'mousetrap/plugins/global-bind/mousetrap-global-bind';
+import { includes } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { useSelect } from '@wordpress/data';
+import { rawShortcut } from '@wordpress/keycodes';
+import { useEffect } from '@wordpress/element';
+
+/**
+ * Return true if platform is MacOS.
+ *
+ * @param {Object} _window   window object by default; used for DI testing.
+ *
+ * @return {boolean}         True if MacOS; false otherwise.
+ */
+function isAppleOS( _window = window ) {
+	const { platform } = _window.navigator;
+
+	return platform.indexOf( 'Mac' ) !== -1 ||
+		includes( [ 'iPad', 'iPhone' ], platform );
+}
+
+/**
+ * Attach a keyboard shortcut handler.
+ *
+ * @param {string} name       Shortcut name.
+ * @param {Function} callback Shortcut callback.
+ * @param {Object} options    Shortcut options.
+ */
+function useShortcut( name, callback, {
+	bindGlobal = false,
+	eventName = 'keydown',
+	target,
+} ) {
+	const { combination, aliases } = useSelect( ( select ) => {
+		return {
+			combination: select( 'core/keyboard-shortcuts' ).getShortcutKeysCombination( name ),
+			aliases: select( 'core/keyboard-shortcuts' ).getShortcutAliases( name ),
+		};
+	}, [ name ] );
+
+	useEffect( () => {
+		const shortcuts = [ combination, ...aliases ].filter( ( shortcut ) => !! shortcut );
```

Comment:
It might be less readable, there's also another option here:

```suggestion
		const shortcuts = [ combination, ...aliases ].filter( Boolean );
```

Or Lodash-y:

```suggestion
		const shortcuts = compact( [ combination, ...aliases ] );
```

Reply:
compact doesn't work for `null` values.

Reply:
> compact doesn't work for `null` values.

It does though?

```
n_ > _.compact( [ null, true, 1, 0, false ] );
[ true, 1 ]
```

Documented to have the same "truthy" filtering behavior you're intending.

https://lodash.com/docs/4.17.15#compact

Reply:
Weird, I promise I was using "compact" initially and noticed bugs. I'll try again :)

Reply:
It works now 🤷‍♂